### PR TITLE
Add file path copy button to PR diffs

### LINKED ISSUE #364: Filename copy method
The ability to toggle the file content during pull requests reviews is awesome, but sometimes the diff is hard to read and when you decide to copy the filename to read it in the IDE, the toggle feature interfere with this process (e.g. when you try to select the filename to copy it, you sometimes toggle the file's content several times).

It is possible to create method to copy the filename in the pull requests to the clipboard?

captura de pantalla 2017-03-05 a la s 17 49 05

Thanks!

### PR DESCRIPTION
**Pull request by AshikNesin**
Added "file path copy" button in pull requests page.

image

Fixes #364

### CODE REVIEW

File: extension/copy-file.js [hunk 1]
```diff
@@ -18,3 +18,32 @@ window.addFileCopyButton = () => {
 		utils.copyToClipboard(fileContents);
 	});
 };
+window.addFilePathCopyButton = () => {
```
Comment (paulmolluzzo) [inline extension/copy-file.js]: Locally this whole section throws an error in `npm test` because of `xo` errors.

File: extension/content.css [hunk 2]
```diff
@@ -586,6 +586,10 @@ div.inline-comment-form .form-actions,
 	float: none;
 }
 
+/* Add "copy filepath" button for PR page */
+.copy-filepath-btn:hover{
+	cursor: pointer;
+}
```
Comment (paulmolluzzo) [inline extension/content.css]: (Nit) Could you please add a line below this block?

File: extension/copy-file.js [hunk 3]
```diff
@@ -18,3 +18,32 @@ window.addFileCopyButton = () => {
...
+    const $files = $('#files .file');
+    $files.each((i, el) => {
+    	// // Button already added
```
Comment (paulmolluzzo) [inline extension/copy-file.js]: Double commenting.

Comment (busches) [inline extension/copy-file.js]: This may be a good idea for a new xo rule?

**Comment by AshikNesin** [comment]
Thanks guys.

My first PR 😄 

@paulmolluzzo What does (Nit) mean?

**Comment by paulmolluzzo** [comment]
@AshikNesin short for "nitpick". Meaning it's a super tiny thing that is really not important but I'm bugging you about it. 😉

File: extension/content.css (hunk 2 above)
Comment (paulmolluzzo) [inline extension/content.css]: @AshikNesin I meant a line between the two blocks so they're separated. This CSS isn't related to the "Add a Project" button CSS

Comment (AshikNesin) [inline extension/content.css]: @paulmolluzzo Seems like I missed that  😞 

Can I send a update this PR?

File: extension/content.js [hunk 4]
```diff
@@ -480,6 +480,9 @@ document.addEventListener('DOMContentLoaded', () => {
 			if (pageDetect.isPRFiles() || pageDetect.isPRCommit()) {
 				diffFileHeader.setup();
 				addDiffViewWithoutWhitespaceOption('pr');
+				addFilePathCopyButton();
+				const $files = $('#files_bucket #files');
+				new MutationObserver(addFilePathCopyButton).observe($files[0], {childList: true, subtree: true});
```
Comment (sindresorhus) [inline extension/content.js]: The mutation observer could be moved into the `addFilePathCopyButton` function.

File: extension/copy-file.js [hunk 5]
```diff
@@ -18,3 +18,33 @@ window.addFileCopyButton = () => {
 		utils.copyToClipboard(fileContents);
 	});
 };
+
+window.addFilePathCopyButton = () => {
```
Comment (sindresorhus) [inline extension/copy-file.js]: Put it in a new file. And remember to add that file in manifest.json

**Comment by sindresorhus** [comment]
Can you add it to the https://github.com/sindresorhus/refined-github#highlights section in the readme with a link to a screenshot?

**Comment by sindresorhus** [comment]
The tooltip seems a bit unaligned. It's too close to the button and a little bit too much to the left.

**Comment by AshikNesin** [comment]
@sindresorhus
Since all the tooltip in that section are positioned top, have changed position of it to be top too (let's keep it consistent 😄 )

Is it okay now?

redefined-github-file-path-copy

**Comment by sindresorhus** [comment]
The tooltip says "Copy to clipboard", but I think we should be explicit about what it's copying. Change it to "Copy file path to clipboard".

I also noticed that the other buttons get a blue highlight when hovered, but this one does not. Can you add that too so it fits in better with the other icon buttons?

**Comment by sindresorhus** [comment]
Excellent work @AshikNesin. Keep the PRs coming! :)

**Comment by AshikNesin** [comment]
@sindresorhus Sure 😄

**Comment by bfred-it** [comment]
Does anyone use this? Is this necessary after [this?](https://github.com/sindresorhus/refined-github/issues/364#issuecomment-300887977)

**Pull request merged**